engine/proofview.ml

```diff
@@ -1211,7 +1211,7 @@ module V82 = struct
    { Evd.it = comb ; sigma = solution }
 
   let top_goals initial { solution=solution; } =
-    let goals = CList.map (fun (t,_) -> fst (Constr.destEvar (EConstr.Unsafe.to_constr t))) initial in
+    let goals = CList.map (fun (t,_) -> fst (EConstr.destEvar solution t)) initial in
```

Comment:
It looks like this change causes a destKO when calling Guard and there is no current goal (all goals solved, or current bullet solved and we're waiting to refocus).